model/model_training/custom_datasets/qa_datasets.py

```diff
@@ -211,7 +212,14 @@ def __init__(self, mode: str = "sft", max_answers: int = 5) -> None:
             # Then take only the first `max_answers` elements (usually there are just
             # 2, but there are examples where we have more)
             answers_sorted = [x[0] for x in sorted(answers.items(), key=lambda x: -1 * x[1])]
-            self.rows.append(DatasetEntry(questions=[question], answers=[answers_sorted[:max_answers]], lang="en"))
+            self.rows.append(
+                create_dataset_entry_qa(
+                    mode=mode,
+                    questions=[question],
+                    answers=[answers_sorted[:max_answers]],
+                    lang="en",
```

Comment:
similar (non-)issue here as outlined above: samples are not always guaranteed to be english

Reply:
These lines are part of the WebGPT loader class. Does WebGPT contain non-english entries?

Reply:
i haven’t looked at the dataset, but i would assume that there will be some non-english entries. same as above: probably not worth worrying about for now.

Reply:
I just reran the check on the dataset and these are the results:
```
defaultdict(int,
            {'en': 28272,
             'af': 73,
             'fr': 34,
             'tl': 9,
             'nl': 30,
             'da': 3,
             'so': 4,
             'fi': 5,
             'ca': 9,
             'it': 7,
             'de': 11,
             'pt': 7,
             'ro': 10,
             'no': 4,
             'cy': 10,
             'es': 2,
             'id': 6,
             'sv': 2,
             'pl': 3})
```
So `99.19651942037121 %` of the datasets are english, according to `langdetect`. I would just keep this language tag here